### DISCUSSION TITLE
Network graph - topology?

### DISCUSSION THREAD

**Discussion by Ophiolith** [SlashScreen/skelerealms#8] [General]
Hey! I'm looking to integrate Skelerealms into my hobby-project, and was wondering about the intended way to use the network graph for representing out-of-scene navigation. My intent is to build a STALKER-inspired game where NPCs have a lot of agency to roam between locations according to goals defined via GOAP. Given this, would the intended use of the network graph be to define points of interest in a map scene represented via nodes? And would portal nodes be used to link between individual map scenes?
I'm sure it's a pretty flexible system, and _intuitively_ that seems like the intended metaphor, but I just wanted to make sure I'm following correctly :)

**Comment by SlashScreen** [comment]
in a sense, yes. How you use it is up to you, but the way I use this is to create lines of nodes along paths/roads and around areas of interest, a bit like a very low-res navmesh. Once an NPC finds and reaches the closest node it can find to the point it wants to go to, it will then move to the point it wants to directly, so you don't need to worry about exactly covering every point your NPCs may want to go to. Also, NPC puppets are programmed to spawn in at the point on the navmesh closest to where they were located before entering the scene, so no need to worry about them getting stuck inside hills or something.

And yes, the "portals" are meant to connect different maps, like the doors. This system will probably be more tightly integrated in the future.

While the navigation within the same world has been tested, navigation between scenes has not been thoroughly tested (Although it *should* work), so if you run into any problems, feel free to open an issue.

**Reply by Ophiolith** [reply]
Brilliant, thank you for the explanation.
